TFNetworkLayer.py

```diff
@@ -5468,14 +5468,18 @@ def init_by_layer(self, layer, layer_output_template=None):
       target=layer._get_target_value(),
       layer=layer)
 
-  def _flatten_or_merge(self, x, seq_lens, time_major):
+  def _flatten_or_merge(self, x, seq_lens, time_major, batch_feature_major=False):
     """
     :param tf.Tensor x: (B,T,...) or (T,B,...)
     :param tf.Tensor seq_lens: (B,)
     :param bool time_major:
+    :param bool batch_feature_major:
     :return: (B*T|T*B|B',...)
     :rtype: tf.Tensor
     """
+    if batch_feature_major:
+      from TFUtil import swapaxes
+      x = swapaxes(x, 1, 2)
```

Comment:
This is wrong in general. You probably mean `-1` instead of `2`.
But also, you are actually swapping it already below, so the whole change here in `_flatten_or_merge` should not be needed?

Reply:
You mean `x = swapaxes(x, 0, 1)`, which goes below? But it's swapping between time_dim and batch_dim.

Reply:
No. I mean what I wrote, I.e. I mean `-1` instead of `2`, i.e. `swapaxes(x, 1, -1)` (which is equivalent to `swapaxes(x, 1, 2)` if `ndim == 3`.